src/events/Windows.cpp

```diff
@@ -667,6 +667,9 @@ void Events::listener_mapWindow(void* owner, void* data) {
 
     if (PMONITOR && PWINDOW->m_iX11Type == 2)
         PWINDOW->m_fX11SurfaceScaledBy = PMONITOR->scale;
+
+    // Move cursor to middle of new window if enabled.
+    g_pCompositor->warpCursorTo(PWINDOW->middle());
```

Comment:
you're not checking if:
- there is a constraint
- the window gained focus
- the window is focusable

Reply:
Good points, especially the checking if its gained focus. I definitely put this out a little per-maturely.

For the first point though, in https://github.com/hyprwm/Hyprland/blob/2e763764bfdaee2039961c6c220b5d6af31712d4/src/Compositor.cpp#L2425-L2443. I see that this is only going to warp if the user hasn't put `general:no_cursor_warps` in their config. I kind of figured it fell under the same bucket as the change focus warps.

I'm going to think on this some more and ideally try to integrate this solution with what changing focus uses.

Thanks for the feedback! Going to put this in Draft for a bit.